### PR TITLE
Added parameters for barometer device IDs

### PR DESCRIPTION
**Pull request by tridge**
This allows us to tell what barometers were present in a log

### CODE REVIEW

File: libraries/AP_Baro/AP_Baro_Backend.h [hunk 1]
```diff
@@ -23,6 +23,28 @@ class AP_Baro_Backend
     // If the value further that filtrer_range from mean value, it is rejected.
     bool pressure_ok(float press);
     uint32_t get_error_count() const { return _error_count; }
+
+    /*
+      device driver IDs. These are used to fill in the devtype field
+      of the device ID, which shows up as GND_BARO_ID* parameters to
+      users.
+     */
+    enum DevTypes {
```
Comment (peterbarker) [inline libraries/AP_Baro/AP_Baro_Backend.h:32]: `enum class DevType`?

File: libraries/AP_Baro/AP_Baro_SITL.cpp [hunk 2]
```diff
@@ -20,6 +20,7 @@ AP_Baro_SITL::AP_Baro_SITL(AP_Baro &baro) :
 #if APM_BUILD_TYPE(APM_BUILD_ArduSub)
         _frontend.set_type(_instance, AP_Baro::BARO_TYPE_WATER);
 #endif
+        set_bus_id(_instance, AP_HAL::Device::make_bus_id(AP_HAL::Device::BUS_TYPE_SITL, 0, _instance, DEVTYPE_BARO_SITL));
```
Comment (peterbarker) [inline libraries/AP_Baro/AP_Baro_SITL.cpp:23]: No `set_device_type`?

Comment (tridge) [reply libraries/AP_Baro/AP_Baro_SITL.cpp:23]: > `set_device_type`

we don't have a dev to call it on, make_bus_id() does all fields

File: libraries/AP_Baro/AP_Baro_Backend.h (hunk 1 above)
Comment (tridge) [reply libraries/AP_Baro/AP_Baro_Backend.h:32]: this PR follows the same pattern as compass and INS

Comment (rmackay9) [reply libraries/AP_Baro/AP_Baro_Backend.h:32]: We are mostly using "enum class" in new code though..

**Pull request merged**